### PR TITLE
libflux: add flux_sync_create()

### PR DESCRIPTION
**Pull request by garlick**
This was peeled of of PR #3512.

This adds a convenience function `flux_sync_create()` which returns a future that is fulfilled when the system heartbeat event is published.  The three main users of the heartbeat:  broker overlay network, content-cache, and KVS are switched over to the new interface.

Besides being simpler than directly subscribing to heartbeat messages, this interface allows a maximum time between callbacks  to be imposed using a continuation timeout.  This is handy in case heartbeats are delayed or configured to be too infrequent for a critical task.  In addition, this version allows a minimum time between callbacks to be set as a parameter.  This can protect against thrashing when heartbeats arrive too frequently.

I also dropped the "epoch" from the heartbeat payload and renamed its topic string from `hb` to `heartbeat.pulse`, to avoid squatting on all topic strings beginning with those two letters.

### CODE REVIEW

File: src/common/libflux/sync.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    flux_future_t *f = arg;
+    struct flux_sync *sync = flux_future_aux_get (f, auxkey);
+    uint32_t seq;
+
+    if (flux_msg_authorize (msg, FLUX_USERID_UNKNOWN) < 0
+            || flux_event_decode (msg, NULL, NULL) < 0
+            || flux_msg_get_seq (msg, &seq) < 0
+            || (sync->count > 0 && seq <= sync->seq)
+            || (sync->count > 0 && sync->minimum > 0.
+                                && now - sync->last < sync->minimum)) {
```
Comment (grondo) [inline src/common/libflux/sync.c:80]: Another extra 4 space indent here.. though maybe you are doing that on purpose... ignore if so.

**Review by grondo — COMMENTED** [review]
Just some very minor comments inline.

There are also 2 other places where `hb` is still referenced:
 * `doc/man3/tevent.c` references `"hb"`, though it is only meant as documentation, but might as well update it
 *  By default the `cron` module subscribes to `hb` to defer executed tasks to the next heartbeat. I think only `rc1` needs to be updated  [here](https://github.com/flux-framework/flux-core/blob/master/etc/rc1#L25)

File: doc/man3/flux_sync_create.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Synchronizing Flux internal overhead to the heartbeat can, in theory,
+reduce disruption to bulk synchronous applications.
+
+If *minimum* is greater than zero, it establishes a minimum time
+between fulfillments.  This may be used to protect from thrashing
+if the heartbeat period is set too fast, or if heartbeats arrive close
+to one another due to overlay network congestion.
+
+A maximum time between fulfillments may be established by specifying
+a continuation timeout with ``flux_future_then()``, if desired.
```
[the remark this answers is not in the document]
Comment (garlick) [reply doc/man3/flux_sync_create.rst]: Thanks, upon reading my man page through this morning, I felt it needed a bit more work overall, so I addressed your suggestion, reworded some other things that I felt were a bit unclear, and added an example.

(on the aside: interesting... should returning from a continuation _always_ reset the future?  Something to ponder)

File: src/common/libflux/sync.c (hunk 1 above)
Comment (garlick) [reply src/common/libflux/sync.c]: It was kind of on purpose but I don't know why.   Will fix.

**Comment by garlick** [comment]
Forced a push after addressing @grondo's comments.  Thanks for catching those additional "hb" usages!  The man page example I _had_ fixed once before but think I lost it in a rebase.  The cron one I had completely missed.

**Comment by garlick** [comment]
Er, just realized one of my unit test cases was completely useless, so forced a push without that.  All done playing around now!

**Review by grondo — APPROVED** [review]
LGTM!

**Pull request merged**